### ISSUE TITLE
90 min PPP 2.8

### ISSUE THREAD

**Issue by anish-kum** [anish-kum/PPP-Spark-2025-2026#26]
90 min PPP

**Comment by anish-kum** [comment]
<img width="336" height="484" alt="Image" src="https://github.com/user-attachments/assets/32774e98-e054-4d9f-b22c-530060f6648d" />


What I learned:

- Data Preprocessing: Emphasizes the importance of preprocessing data to improve model performance. This is important for cleaning and interpreting raw data to be suitable for modeling. It's a bit similar in concept to how images are cleaned and sharpened for better resolution.

- Training Strategies: Discusses various strategies for training models effectively, such as adjusting learning rates and using early stopping. This is important for debugging, especially the early stopping strategy, as these acts as breakpoints to figure out points of inefficiency or slower learning.

- Model Evaluation Metrics: Introduces key metrics for evaluating model performance, including accuracy, precision, etc. This is important for finding faults within a model and identifying those underperforming areas. There are literally jobs that entail this, as tokens for AI prompts are super expensive and the server rooms used for AI are very resource-intensive, so faults that may compromise efficiency NEED to be caught.

- Hyperparameter Tuning: Highlights the significance of tuning hyperparameters to optimize model performance and achieve better results. This is super important for making more accurate predictions and ensuring more efficient resource utilization when performing calculations.


2 Questions I still had:

- How do you know whether poor model performance is caused by bad preprocessing versus poorly tuned hyperparameters?

- When using early stopping, how can you tell if the model is truly finished learning or just temporarily plateauing due to the learning rate?


From your previous feedback, I included more details on what I learned about why this is important/more elaboration.

**Comment by drewmca-pkwy** [comment]
@anish-kum Looks good, Anish. I appreciate the details about what you learned and why it's important to you.
